src/network/networktool.cpp

```diff
@@ -38,19 +38,38 @@ void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
 {
     const static QString projectTag = QStringLiteral(GIT_TAG);
 
+    auto semverToInt = [](const QString& version) -> int {
+        int result = 0;
+        QStringList subVersion = version.right(1).split('.');
+
+        // Check if subVersion is valid
+        if(subVersion.length() > 3)
+        {
+            return result;
+        }
+
+        // Transform version to int
+        for(int i = 0; i < subVersion.length(); i++)
+        {
+            result += subVersion[i].toInt()<<(3-i);
+        }
+        return result;
+    };
+
     // Check if version is a normal release
     // For everything else we can presume test release
-    static QRegularExpression releaseTagRegex(QStringLiteral(R"([v]\d+\.\d+)"));
+    // This regex will check for vX, vX.X and vX.X.X
```

Comment:
This <s>does not seem necessary at all</s> seems wrong to check for anything but vx.x.x.

When the goal is to make a consistent standard, it does not make sense to compare version x with x.x.x. We will never do it. It's like unused code.